cmd/spire-server/cli/validate/validate.go

```diff
@@ -0,0 +1,74 @@
+package validate
+
+import (
+	"flag"
+
+	"github.com/mitchellh/cli"
+	"github.com/spiffe/spire/cmd/spire-server/cli/run"
+	common_cli "github.com/spiffe/spire/pkg/common/cli"
+)
+
+const defaultConfigPath = "server.conf"
+
+func NewValidateCommand() cli.Command {
+	return newValidateCommand(common_cli.DefaultEnv)
+}
+
+func newValidateCommand(env *common_cli.Env) *validateCommand {
+	return &validateCommand{
+		env: env,
+	}
+}
+
+type validateCommand struct {
+	env *common_cli.Env
+
+	configPath string
+	ExpandEnv  bool
+}
+
+func (c *validateCommand) Help() string {
+	// ignoring parsing errors since "-h" is always supported by the flags package
+	_ = c.parseFlags([]string{"-h"})
+	return ""
+}
+
+func (c *validateCommand) Synopsis() string {
+	return "Validates a SPIRE server configuration file"
+}
+
+func (c *validateCommand) Run(args []string) int {
+	if err := c.parseFlags(args); err != nil {
+		return 1
```

Comment:
Do we want to print the error? Does `flag` do that for us?

Reply:
Yes. For example, if the value passed to `-expandEnv` is not bool, the`flag` package prints a parse error as below:
```shell
$ bin/spire-server validate -expandEnv=not-bool
invalid boolean value "not-bool" for -expandEnv: parse error
Usage of validate:
  -config string
        Path to a SPIRE server configuration file (default "server.conf")
  -expandEnv
        Expand environment variables in SPIRE config file

$ echo $?
1
```